File: BackEnd/DotNet/WebDigital/WebDigital/Services/Implements/PublisherService.cs
using Microsoft.EntityFrameworkCore;
using WebDigital.dto;
using WebDigital.Models;
using WebDigital.Services.Interface;

namespace WebDigital.Services.Implements
{
    public class PublisherService : IPublisherService
    {
        private readonly BookwormfinalBackupContext _context;

        public PublisherService(BookwormfinalBackupContext context)
        {
            _context = context;
        }

        public async Task<List<PublisherDto>> GetAll()
        {
            return await _context.Publishers
                .OrderBy(p => p.Name)
                .Select(p => new PublisherDto
                {
                    PublisherId = p.PublisherId,
                    Name = p.Name,
                    Email = p.Email
                })
                .ToListAsync();
        }

        public async Task<int> Create(PublisherCreateDto dto)
        {
            var publisher = new Publisher
            {
                Name = dto.Name,
                Email = dto.Email
            };

            _context.Publishers.Add(publisher);
            await _context.SaveChangesAsync();

            return publisher.PublisherId;
        }
    }
}

File: BackEnd/Java/product-service/src/main/java/com/digitalshelf/productservice/service/PublisherService.java
package com.digitalshelf.productservice.service;

import com.digitalshelf.productservice.dto.PublisherCreateDto;
import com.digitalshelf.productservice.dto.PublisherDto;
import com.digitalshelf.productservice.entity.Publisher;
import com.digitalshelf.productservice.repository.PublisherRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Comparator;
import java.util.List;

@Service
@RequiredArgsConstructor
public class PublisherService {

    private final PublisherRepository publisherRepository;

    public List<PublisherDto> getAll() {
        return publisherRepository.findAll().stream()
                .sorted(Comparator.comparing(Publisher::getName))
                .map(p -> PublisherDto.builder().publisherId(p.getPublisherId()).name(p.getName()).email(p.getEmail()).build())
                .toList();
    }

    public Integer create(PublisherCreateDto dto) {
        Publisher publisher = new Publisher();
        publisher.setName(dto.getName());
        publisher.setEmail(dto.getEmail());
        publisherRepository.save(publisher);
        return publisher.getPublisherId();
    }
}